rgsync/Connectors/redis_connector.py

```diff
@@ -56,81 +100,108 @@ def WriteData(self, data):
-
-                if op == OPERATION_UPDATE_REPLICATE:
-                    d = [[k, v] for k,v in d_val.items() if not k.startswith('_')]
-                    res = self.session.execute_command('HSET', newKey, *sum(d, []))
-                    if 'ERR' in str(res):
-                        WriteBehindLog("Error: Command HSET execution failed, error = %s" % str(res))
-                        raise Exception(res) from None 
+                newKey = '{}:{}'.format(self.new_prefix, pk)
+
+                self.shouldCompareId = False
+                if op != OPERATION_UPDATE_REPLICATE:
+                    if isAddBatch:
+                        # process update batch
+                        for kv_pair in batch:
+                            for k, v in kv_pair.items():
+                                pipe.hset(k, mapping=v)
+                        
+                        batch = []
+                        isAddBatch = False
+
+                    # append key to delete in list
+                    batch.append(newKey)
                 else:
-                    res = self.session.execute_command('DEL', newKey)
-                    if 'ERR' in str(res):
-                        WriteBehindLog("Error: Command DEL execution failed, error = %s" % str(res))
-                        raise Exception(res) from None
-                    
-                # make entry for exactly once
-                if self.exactlyOnceTableName is not None:
-                    res = self.session.execute_command('HSET', self.exactlyOnceTableName, shardId, lastStreamId)
-                    if 'ERR' in str(res):
-                        WriteBehindLog("Error: Entry for exactly once failed key: %s, field: %s, value: %s" 
-                                % (self.exactlyOnceTableName, shardId, lastStreamId))
-                        WriteBehindLog("Error: Command HSET execution failed, error = %s" % str(res))
-                        raise Exception(res) from None
+                    if not isAddBatch:
+                        # process delete batch
+                        if(isinstance(self.connection, RedisConnection)):
+                            pipe.delete(*batch)
+                        elif(isinstance(self.connection, RedisClusterConnection)):      # multi key deletion is not supported in cluster
+                            for key in batch:
+                                pipe.delete(key)
+
+                        batch = []
+                        isAddBatch = True
+
+                    # make a dictionary and adda element with new key as its key and field-value mapping as its value
+                    # append this dictionary to list
+                    d = dict()
+                    d[newKey] = d_val
+                    batch.append(d)
+
+            if(0 < len(batch)) and isAddBatch:
+                for kv_pair in batch:
+                    for k, v in kv_pair.items():
+                        pipe.hset(k, mapping=v)
```

Comment:
@ratnesh-tiwari why not adding the command to the pipe directly? why you need to add them to batch list?